### PR TITLE
Add app engine template

### PR DESCRIPTION
**Pull request by sourced-vince**

### CODE REVIEW

**Comment by fawix** [comment]
We need to clarify instructions for user. The end-goal is to have their app deployed by the template when starting from zero (think a brand new project). 

The user should not need to run `app deploy` at the end of our template run. 

Today when I ran this template it does not seem to do anything (unless I am missing some steps) - although I do see the deployment under DM there is no app engine dashboard and all services and submenus seem empty -  which tells me something is not right.

Here is what I think we are missing:
- [ ] Would be helpful to include a note about where his code should sit in relation to the template. 
- [ ] Allow the user to select between Standard and Flexible environments
- [ ] Allow the user to select the runtime 
- [ ] Add `cookie expiration` as optional field
- [ ] Add `gcr domain` as optional field
- [ ] Create a functional example that deploys a hello world (can use it from the samples repo)
- [ ] Document in the example the pre-requisite steps to run it.

https://cloud.google.com/appengine/downloads?hl=en_US&_ga=2.213576980.-1649752503.1535218256

We can discuss in further detail if need.

File: community/cloud-foundation/templates/app_engine/tests/integration/app_engine.bats [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "test-app-engine-flexible-${RAND}" ]]
+
+    run gcloud app versions describe "test-app-engine-flexible-${RAND}" \
+        --project "${CLOUD_FOUNDATION_PROJECT_ID}" \
+        --service=default
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "test-app-engine-flexible-${RAND}" ]]
+    [[ "$output" =~ "manualScaling:" ]]
+    [[ "$output" =~ "instances: 5" ]]
```
Comment (fawix) [inline community/cloud-foundation/templates/app_engine/tests/integration/app_engine.bats]: we should also validate `env: flexible`

File: community/cloud-foundation/templates/app_engine/tests/integration/app_engine.bats [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+@test "Verifying resource was created in deployment ${DEPLOYMENT_NAME}" {
+    run gcloud app describe --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "id: ${CLOUD_FOUNDATION_PROJECT_ID}" ]]
+    [[ "$output" =~ "locationId: us-east1" ]]
+    [[ "$output" =~ "servingStatus: SERVING" ]]
+}
+
+@test "Verifying that flexible app engine resource were created in deployment ${DEPLOYMENT_NAME}" {
```
Comment (fawix) [inline community/cloud-foundation/templates/app_engine/tests/integration/app_engine.bats]: Can we also add a standard version to the same app engine to test standard is being created ok?

**Comment by fawix** [comment]
Hi Vince,

When running the flex example I get a funny issue with Cloud Build; any idea? 
```
{"ResourceType":"appengine.v1.version","ResourceErrorCode":"500","ResourceErrorMessage":"Cloud build 5584b6de-5f6b-498c-b59a-338c56171d5f status: FAILURE"}
```

When running the integration tests I get two failures:

```
 ✗ Verifying standard app engine resource were created in deployment copper-booster-224502-app-engine-rsnsf1zqem
   (in test file templates/app_engine/tests/integration/app_engine.bats, line 72)
     `[[ "$output" =~ "test-ae-std-${RAND}" ]]' failed
 ✗ Verifying flex app engine resource were created in deployment copper-booster-224502-app-engine-rsnsf1zqem
   (in test file templates/app_engine/tests/integration/app_engine.bats, line 86)
     `[[ "$output" =~ "test-ae-flex-${RAND}" ]]' failed
   Deleting .copper-booster-224502-app-engine-rsnsf1zqem.yaml
```

**Comment by fawix** [comment]
PS: please ignore the close I hit the wrong button! sorry!

**Comment by sourced-vince** [comment]
> Hi Vince,
> 
> When running the flex example I get a funny issue with Cloud Build; any idea?
> 
> ```
> {"ResourceType":"appengine.v1.version","ResourceErrorCode":"500","ResourceErrorMessage":"Cloud build 5584b6de-5f6b-498c-b59a-338c56171d5f status: FAILURE"}
> ```
> When running the integration tests I get two failures:
> 
> ```
>  ✗ Verifying standard app engine resource were created in deployment copper-booster-224502-app-engine-rsnsf1zqem
>    (in test file templates/app_engine/tests/integration/app_engine.bats, line 72)
>      `[[ "$output" =~ "test-ae-std-${RAND}" ]]' failed
>  ✗ Verifying flex app engine resource were created in deployment copper-booster-224502-app-engine-rsnsf1zqem
>    (in test file templates/app_engine/tests/integration/app_engine.bats, line 86)
>      `[[ "$output" =~ "test-ae-flex-${RAND}" ]]' failed
>    Deleting .copper-booster-224502-app-engine-rsnsf1zqem.yaml
> ```

The 500 indicates an internal error. If you check the App Engine dash, you will notice that the app is properly deployed. Anyone at Google who might have an idea?

**Pull request merged**